### DISCUSSION TITLE
WASM without Razor

### DISCUSSION THREAD

**Discussion by samcov** [dotnet/aspnetcore#24421] [General] [answered]
Is it possible to simply compile C# to wasm and execute it from javascript, of course passing data both ways, _without_ using Razor?

I ask because I have no interest in Razor, I just want to use C# in the browser for my own evil purposes :) 

I didn't see any sample code for this use case.

**Comment by hultqvist** [comment, marked as answer]
It's been discussed in #16021
Blazor is using mono to compile c# into webassembly, you should be able to use mono to compile to wasm.

**Reply by samcov** [reply]
Thanks, that really clears it up for me, but unfortunately, it's the same old story.

The most fundamental thing associated with getting to use C# in the browser is simply ignored as a use case.

Even worse, the lengths many people have to go through for people that should "Get it" to understand this simple request is astounding, but in my experience normal.

When Windows 8 came out, I suggested that IE have "Paste and Go" as an option, and couldn't get anyone on the team to recognize why that was important or needed. Fast forward to today, no browser is without it(well, except old IE's, lol).
